gateway/src/apicast/policy/caching/policy.lua

```diff
@@ -38,13 +44,37 @@ local function resilient_handler(cache, cached_key, response, ttl)
   end
 end
 
+local function handle_500_allow_mode(cache, cached_key, ttl)
```

Comment:
Does it warrant own function now?

Reply:
I don't have a strong preference. I'm inclined to say yes because of the long comment.